src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -2396,7 +2397,11 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
 
     node_to_enter           <= inputs.enter_selected_node.map(f_!(model.last_selected_node()));
     out.source.node_entered <+ node_to_enter;
+    removed_edge            <= out.node_entered.map(f_!(model.model.clear_all_detached_edges()));
+    out.source.edge_removed <+ removed_edge;
     out.source.node_exited  <+ inputs.exit_node;
+    removed_edge            <= out.node_exited.map(f_!(model.model.clear_all_detached_edges()));
+    out.source.edge_removed <+ removed_edge;
```

Comment:
In FRP, always try to merge the flows as early as possible. thus instead of attaching two separate streams to `out...` its always better to design the streams as:
```rust
removed_edge_on_enter <- ...
removed_edge_on_exit   <- ...
removed_edge <- any(&removed_edge_on_enter,&removed_edge_on_exit);
out.source.edge_removed <+ removed_edge;
```

We are not always doing it in graph editor, as the whole FRP definition is huge and is meant ot be split to several functions, but if you are working locally - like only on edge connection, deteach etc, you should design it always to mergethe streams as early as possible, please :)